Assets/Scripts/Interactables/CocktailMiniGame.cs

```diff
@@ -0,0 +1,374 @@
+            else if (isMiniGameActive)
+            {
+                // Only end the mini-game if it was active
+                EndCocktailMiniGame(false);
+            }
+        }
+
+        //CocktailMiniGame Controller, similar to beer
+        public override void Interact(PlayerInteraction player) 
+        {
+            base.Interact(player);
+            this.player = player; //store the player ref
+
+            // Checks to see if player could receive a beer at end of interaction
+            if (player.CurrentInventory.CanAddItem(itemToSpawn)) //must add detection to see if cx has ingredients
+            {
+              //  GameManager.Instance.UpdateGameState(GameManager.GameState.CocktailMiniGame); - legacy code, seeiong if other option works
+
+                GameManager.Instance.StartCocktailMiniGame();
+            }
+
+        }
+
+        private void StartCocktailMiniGame() //rythm mini game
+        {
+            isMiniGameActive = true;
+
+            detectedInputDuringMiniGame = false;
+
+            // Hide or show relevant elements for the minigame
+            if (cocktailShakerRenderer != null)
+            {
+                cocktailShakerRenderer.enabled = true; // Ensure it remains visible during gameplay
+            }
+
+            // Set initial left arrow sprite
+            if (leftArrow != null) 
+            { 
+                leftArrow.SetActive(true);
+                leftArrow.GetComponent<Image>().sprite = leftUIUnpressed;
+            }
+
+            // Set initial right arrow sprite
+            if (rightArrow != null)
+            {
+                rightArrow.SetActive(true);
+                rightArrow.GetComponent<Image>().sprite = rightUIUnpressed;
+            }
+
+            // Enable the Canvas for the UI arrows
+            if (cocktailCanvasGroup != null)
+            {
+                cocktailCanvasGroup.alpha = 1f;
+                cocktailCanvasGroup.interactable = true;
+                cocktailCanvasGroup.blocksRaycasts = true;
+            }
+
+            rhythmCoroutine = StartCoroutine(SwitchArrowsCoroutine());
+
+            // Start the timer coroutine for automatic end
+            StartCoroutine(MiniGameTimer());
+        }
+
+        private IEnumerator SwitchArrowsCoroutine()
+        {
+            while (GameManager.Instance.State == GameManager.GameState.CocktailMiniGame)
+            {
+                if (Input.GetKeyDown(Quit))
+                {
+                    Debug.Log("Quit key pressed");
+                    EndCocktailMiniGame(false);
+                    yield break; // Exit the coroutine
+                }
+
+                // Alternate between left and right arrow
+                leftArrowActive = !leftArrowActive;
+
+                // Switch the sprite based on active arrow
+                //Sets the sprite property of the Image component on each arrow to either the pressed or unpressed state
+                if (leftArrowActive)
```

Comment:
null checks should no longer be needed. We should confirm null checks early, and if anything still null, just not let minigame happen.